### PR TITLE
[BUGS#459] fix: check URL before doWikiGet

### PR DESCRIPTION
**Pull request by miurahr**
- Resolved BUGS#459 Print Exception with Invalid URL in "Import From MediaWiki..

## Pull request type

Please mark github LABEL of the type of change your PR introduces:

- Bug fix -> [bug]

## Which ticket is resolved?

 
- Print Exception with Invalid URL in "Import From MediaWiki..
- https://sourceforge.net/p/omegat/bugs/459/

## What does this PR change?

- Extend HttpConnectionUtils to have checkUrl method
- Refactor JTextPaneLinkifier to move URL_PATTERN to HttpConnectionUtils class

### CODE REVIEW

**Comment by brandelune** [comment]
I don't understand the dialog when the URL is supposed to be invalid
Screen Shot 2023-10-27 at 8 52 23

If that means the user can still use the URL, we should have:

```
The provided URI looks invalid. Do you still want to use it?
[Yes] [No]
```

There is no need for [Cancel].

If the user hits [Yes] and the URL was really invalid, then there should be a dialog like this:

```
The provided URI was invalid.
[OK]
```

File: src/org/omegat/gui/main/ProjectUICommands.java [hunk 1]
```diff
@@ -1307,6 +1307,11 @@ public static void doWikiImport() {
             return;
         }
         try {
+            if (!HttpConnectionUtils.checkUrl(remoteUrl)) {
+                JOptionPane.showConfirmDialog(Core.getMainWindow().getApplicationFrame(),
+                        OStrings.getString("TF_WIKI_IMPORT_URL_ERROR"));
```
Comment (miurahr) [inline src/org/omegat/gui/main/ProjectUICommands.java:1312]:
```suggestion
                JOptionPane.showMessageDialog(Core.getMainWindow().getApplicationFrame(),
                        OStrings.getString("TF_WIKI_IMPORT_URL_ERROR"));
```

Comment (miurahr) [reply src/org/omegat/gui/main/ProjectUICommands.java:1312]: @brandelune  Is it what you want to say?
https://docs.oracle.com/javase/tutorial/uiswing/components/dialog.html

Comment (brandelune) [reply src/org/omegat/gui/main/ProjectUICommands.java:1312]: The question is: do we allow users to try an invalid URI or not?

If yes, the current dialog must be adapted and there should be a second dialog that says whether the resource existed or not.

If no, then the dialog you now suggest is what we need.

Comment (miurahr) [reply src/org/omegat/gui/main/ProjectUICommands.java:1312]: When allowd invalid URL, you can enjoy exception stack trace.

File: src/org/omegat/Bundle.properties [hunk 2]
```diff
@@ -286,6 +286,7 @@ TF_FILE_IMPORT_TITLE=Add Files to Source Folder...
 TF_MENU_WIKI_IMPORT=Add MediaWiki Pa&ge...
 TF_WIKI_IMPORT_TITLE=Add MediaWiki Page
 TF_WIKI_IMPORT_PROMPT=URL
+TF_WIKI_IMPORT_URL_ERROR=Provided URL is invalid.
```
Comment (miurahr) [inline src/org/omegat/Bundle.properties:289]:
```suggestion
TF_WIKI_IMPORT_URL_ERROR=Provided URL is invalid.
TF_WIKI_IMPORT_URL_ERROR_TITLE=Url Error
```

File: src/org/omegat/gui/main/ProjectUICommands.java (hunk 1 above)
Comment (brandelune) [reply src/org/omegat/gui/main/ProjectUICommands.java:1312]: Ok.

File: src/org/omegat/Bundle.properties (hunk 2 above)
Comment (brandelune) [reply src/org/omegat/Bundle.properties:289]: Do we use URL or URI ?

Comment (miurahr) [reply src/org/omegat/Bundle.properties:289]: Input dialog use "URL" for prompt.

```
TF_MENU_WIKI_IMPORT=Add MediaWiki Pa&ge...
TF_WIKI_IMPORT_TITLE=Add MediaWiki Page
TF_WIKI_IMPORT_PROMPT=URL
```

Comment (brandelune) [reply src/org/omegat/Bundle.properties:289]: Yes. I thought URI was a better choice for our UI in general but that does not seem to be the case. So let's leave URL.

**Pull request merged**